crates/electrum/src/electrum_ext.rs

```diff
@@ -140,21 +114,17 @@ impl<K> ElectrumUpdate<K, ConfirmationHeightAnchor> {
             }
         };
 
-        Ok(WalletUpdate {
-            last_active_indices: update.last_active_indices,
-            graph: {
-                let mut graph = TxGraph::default();
-                graph.apply_changeset(graph_changeset);
-                graph
-            },
-            chain: update.chain,
-        })
+        let mut new_graph = TxGraph::default();
+        new_graph.apply_changeset(graph_changeset);
+        Ok(new_graph)
     }
 }
 
+type ElectrumUpdate = (local_chain::Update, RelevantTxids);
```

Comment:
I think it would be better if this was a struct and explain why the update is in this form (we have to update the chain and the txids at the same time since we anchor the txids to the same chain tip that we check before and after we gather the txids).